src/Symfony/Component/Ldap/Entry.php

```diff
@@ -36,6 +36,18 @@ public function getDn()
     }
 
     /**
+     * Returns whether an attribute exists.
+     *
+     * @param $name string The name of the attribute
+     *
+     * @return bool
+     */
+    public function hasAttribute($name)
+    {
+        return isset($this->attributes[$name]);
```

Comment:
Using a real world (though fake data) LDAP octet stream of length zero, internally the entry contains a value which is an empty PHP string. As far as I can find any non-existing attributes do not return in the entry, and null values get converted to empty strings.

Output:
```
// var_dump($entry)
object(Symfony\Component\Ldap\Entry)#23 (2) {
  ["dn":"Symfony\Component\Ldap\Entry":private]=>
  string(34) "uid=john2,ou=People,dc=localdomain"
  ["attributes":"Symfony\Component\Ldap\Entry":private]=>
  array(21) {
    ["objectClass"]=>
    array(4) {
      ["count"]=>
      int(3)
      [0]=>
      string(13) "inetOrgPerson"
      [1]=>
      string(12) "posixAccount"
      [2]=>
      string(13) "shadowAccount"
    }
    [0]=>
    string(11) "objectClass"
    ["uid"]=>
    array(2) {
      ["count"]=>
      int(1)
      [0]=>
      string(5) "john2"
    }
    ...
    [6]=>
    string(9) "gidNumber"
    ["gecos"]=>
    array(2) {
      ["count"]=>
      int(1)
      [0]=>
      string(0) ""
    }
    [7]=>
    string(5) "gecos"
    ["loginShell"]=>
    array(2) {
      ["count"]=>
      int(1)
      [0]=>
      string(9) "/bin/bash"
    }
    [8]=>
    string(10) "loginShell"
    ["homeDirectory"]=>
    array(2) {
      ["count"]=>
      int(1)
      [0]=>
      string(10) "/home/john"
    }
    [9]=>
    string(13) "homeDirectory"
    ["count"]=>
    int(10)
  }
}

// var_dump($entry->hasAttibute('gecos'))
bool(true)

// var_dump($entry->getAttribute('gecos'))
array(2) {
  ["count"]=>
  int(1)
  [0]=>
  string(0) ""
}
```

So there should be no problem, because empty strings return an isset of `true`.